bpython/test/test_autocomplete.py

```diff
@@ -145,3 +145,38 @@ def test_formatting_takes_just_last_part(self):
         self.assertEqual(self.completer.format('/hello/there/'), 'there/')
         self.assertEqual(self.completer.format('/hello/there'), 'there')
 
+class MockNumPy(object):
+    """
+        This is a mock numpy object that raises an error when it is trying to be converted to a boolean.
+    """
+    def __nonzero__(self):
+        raise ValueError("The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()")
+
+
+class TestDictKeyCompletion(unittest.TestCase):
+
+    def test_set_of_keys_returned_when_matches_found(self):
+        com = autocomplete.DictKeyCompletion()
+        local={'d':{"ab":1, "cd":2},}
+        result=com.matches(2,"d[" , local)
+        self.assertSetEqual(result,set(["'ab'","'cd'"]))
+
+    def test_empty_set_returned_when_eval_error(self):
+        com = autocomplete.DictKeyCompletion()
+        local={'e':{"ab":1, "cd":2},}
+        result=com.matches(2,"d[" , local)
+        self.assertSetEqual(result,set())
+
+    def test_empty_set_returned_when_not_dict_type(self):
+        com = autocomplete.DictKeyCompletion()
+        local={'l':["ab", "cd"],}
+        result=com.matches(2,"l[" , local)
+        self.assertSetEqual(result,set())
+
+    def test_obj_that_does_not_allow_bool_tests(self):
+        com = autocomplete.DictKeyCompletion()
+        local={'mNumPy':MockNumPy(),}
+        try:
+            com.matches(7,"mNumPy[" , local)
+        except ValueError:
+            raise AssertionError("Dict key completion raised value error.")
```

Comment:
Just let the `ValueError` pass. If a test throws anything, it is considered as failure. I'd prefer a `self.asserSetEqual(...)` as above.